dm/worker/subtask.go

```diff
@@ -186,36 +193,59 @@ func (st *SubTask) Run() {
 }
 
 func (st *SubTask) run() {
-	st.setStage(pb.Stage_Paused)
-	err := st.unitTransWaitCondition()
+	st.setStage(pb.Stage_Running)
+	ctx, cancel := context.WithCancel(st.ctx)
+	st.setCurrCtx(ctx, cancel)
+	err := st.unitTransWaitCondition(ctx)
 	if err != nil {
 		st.l.Error("wait condition", log.ShortError(err))
 		st.fail(err)
 		return
+	} else if ctx.Err() != nil {
+		return
```

Comment:
Will the stage keep as `Stage_Running`?

Reply:
`ctx.Err() != nil` means this context is canceled in other go routine, that go routine will change the stage.

Reply:
So, how about checking the stage in the unit test cases?